File: CS/GaugePropertyEditor.Win/StateIndicatorGaugePropertyEditor.cs
using System;
using System.Drawing;
using DevExpress.XtraGauges.Win;
using DevExpress.ExpressApp.Editors;
using DevExpress.XtraGauges.Core.Model;
using DevExpress.XtraGauges.Win.Gauges.State;

namespace GaugePropertyEditor.Win {
    [PropertyEditor(typeof(Enum), false)]
    public class StateIndicatorGaugePropertyEditor : WinGaugePropertyEditorBase {
        public StateIndicatorGaugePropertyEditor(Type objectType, DevExpress.ExpressApp.Model.IModelMemberViewItem info)
            : base(objectType, info) {
        }
        public override void SetupGaugeCore(GaugeControl gaugeContainer) {
            StateIndicatorGauge siGauge = gaugeContainer.AddStateIndicatorGauge();
            StateIndicatorComponent stateIndicator = siGauge.AddIndicator();
            BindableComponent = stateIndicator;
            StateIndicatorShapeType[] shapes = new StateIndicatorShapeType[] {
                        StateIndicatorShapeType.TrafficLight2,
                        StateIndicatorShapeType.TrafficLight3,
                        StateIndicatorShapeType.TrafficLight4,
                    };
            stateIndicator.States.Clear();
            foreach (StateIndicatorShapeType shape in shapes) {
                IndicatorState state = new IndicatorState();
                state.ShapeType = shape;
                stateIndicator.States.Add(state);
            }
            stateIndicator.Size = new SizeF(100, 200);
        }
        protected override object GetControlValueCore() {
            return ((StateIndicatorComponent)BindableComponent).StateIndex;
        }
        protected override void ReadValueCore() {
            ((StateIndicatorComponent)BindableComponent).StateIndex = Convert.ToInt32(PropertyValue);
        }
    }
}

File: VB/GaugePropertyEditor.Win/StateIndicatorGaugePropertyEditor.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports DevExpress.XtraGauges.Win
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.XtraGauges.Core.Model
Imports DevExpress.XtraGauges.Win.Gauges.State

Namespace GaugePropertyEditor.Win
    <PropertyEditor(GetType(System.Enum), False)> _
     Public Class StateIndicatorGaugePropertyEditor
        Inherits WinGaugePropertyEditorBase
        Public Sub New(ByVal objectType As Type, ByVal info As DevExpress.ExpressApp.Model.IModelMemberViewItem)
            MyBase.New(objectType, info)
        End Sub
        Public Overrides Sub SetupGaugeCore(ByVal gaugeContainer As GaugeControl)
            Dim siGauge As StateIndicatorGauge = gaugeContainer.AddStateIndicatorGauge()
            Dim stateIndicator As StateIndicatorComponent = siGauge.AddIndicator()
            BindableComponent = stateIndicator
            Dim shapes() As StateIndicatorShapeType = {StateIndicatorShapeType.TrafficLight2, StateIndicatorShapeType.TrafficLight3, StateIndicatorShapeType.TrafficLight4}
            stateIndicator.States.Clear()
            For Each shape As StateIndicatorShapeType In shapes
                Dim state As New IndicatorState()
                state.ShapeType = shape
                stateIndicator.States.Add(state)
            Next shape
            stateIndicator.Size = New SizeF(100, 200)
        End Sub
        Protected Overrides Function GetControlValueCore() As Object
            Return (CType(BindableComponent, StateIndicatorComponent)).StateIndex
        End Function
        Protected Overrides Sub ReadValueCore()
            CType(BindableComponent, StateIndicatorComponent).StateIndex = Convert.ToInt32(PropertyValue)
        End Sub
    End Class
End Namespace